src/layers/GeomodelLabelsLayer.ts

```diff
@@ -116,6 +102,10 @@ export class GeomodelLabelsLayer extends CanvasLayer {
       return;
     }
 
+    if (!this.areasWithAvgTopDepth) {
+      this.generateSurfacesWithAvgDepth();
+    }
```

Comment:
This will not recalculate for new data. Need to clear areasWithAvgTopDepth on setData() or similar.

Reply:
My bad, will have a look at this on monday

Reply:
Sidenote: do you guys know if there are issues with this in subclasses when called from constructor? The previous code set this.areasWithAvgTopDepth from the result of old generateSurfacesWithAvgDepth() in setData() called from constructor, I console logged and this.areasWithAvgTopDepth contained data however console.log(this) on the next line showed no data in member areasWithAvgTopDepth. I know this can be off sometimes (bind/classes and all that) but I would think it would be the same object inside a function at there very least.

Reply:
No, don't think we have experienced that. But might be that we always set data after constructor. 

Maybe moving the stuff to onUpdate instead of setData can avoid some potential issues. It is a little convoluted now, and we do some weird things like setting this._data both in setData() and onUpdate even though one uses this._data = parameter and the other this._data = parameter.data

And the constructor of GeomodelLabelsLayer calles super() before it binds it's members. Don't think that should mess it up though.